odl/tomo/util/trafos.py

```diff
@@ -0,0 +1,248 @@
+# Copyright 2014, 2015 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+# Imports for common Python 2/3 codebase
+from __future__ import (print_function, division, absolute_import,
+                        unicode_literals)
+from math import sin, cos, atan2, acos, pi
+from future import standard_library
+
+standard_library.install_aliases()
+from builtins import int
+
+# External
+import numpy as np
+
+
+def euler_matrix(*angles):
+    """Rotation matrix in 2 and 3 dimensions.
+
+    Compute the Euler rotation matrix from angles given in radians.
+    Its rows represent the canonical unit vectors as seen from the
+    rotated system while the columns are the rotated unit vectors as
+    seen from the canonical system.
+
+    Parameters
+    ----------
+    angle1,...,angleN : RealNumber
+        One angle results in a (2x2) matrix representing a
+        counter-clockwise rotation. Two or three angles result in a
+        (3x3) matrix and are interpreted as Euler angles of a 3d
+        rotation according to the 'ZXZ' rotation order, see the
+        Wikipedia article `Euler angles`_.
+
+    Returns
+    -------
+    mat : `numpy.matrix`, shape `(2, 2)` or `(3, 3)`
+        The rotation matrix
+
+    .. _Euler angles:
+        https://en.wikipedia.org/wiki/Euler_angles#Rotation_matrix
+    """
+    dim = 3
+
+    if len(angles) == 1:
+        phi = angles[0]
+        theta = psi = 0.
+        dim = 2
+    elif len(angles) == 2:
+        phi = angles[0]
+        theta = angles[1]
+        psi = 0.
+    elif len(angles) == 3:
+        phi = angles[0]
+        theta = angles[1]
+        psi = angles[2]
+    else:
+        raise ValueError('Number of angles must be between 1 and 3')
+
+    cph = cos(phi)
+    sph = sin(phi)
+    cth = cos(theta)
+    sth = sin(theta)
+    cps = cos(psi)
+    sps = sin(psi)
+
+    if dim == 2:
+        mat = np.matrix([[cph, -sph], [sph, cph]])
+    else:
+        mat = np.matrix([[cph * cps - sph * cth * sps,
```

Comment:
Would make sense. But what name should one use? `trafos`, `rotations`, or simply put it in `utility`?